kano_updater/utils.py

```diff
@@ -677,3 +679,58 @@ def disable_power_button():
     except Exception:
         # Kano Peripherals doesn't support this function
         pass
+
+
+def verify_kit_is_plugged():
+    """
+    On Computer Kit 2 Pro, verify that the battery is plugged in and not depleting.
+    For now, we rely on the user to tell us.
+
+    Returns:
+        is_plugged - bool whether the kit is plugged in or not.
+    """
+    ck2_pro = False
+    is_battery_low = False
+    is_plugged = True
+
+    try:
+        from kano_peripherals.wrappers.detection import is_ck2_pro
+        ck2_pro = is_ck2_pro()
+
+        from kano_peripherals.ck2_pro_hat.driver.high_level import get_ck2_pro_hat_interface
+        ck2pro_iface = get_ck2_pro_hat_interface()
+        is_battery_low = (ck2pro_iface and ck2pro_iface.is_battery_low())
+    except:
+        # Kano Peripherals doesn't support this function
+        pass
```

Comment:
To be honest, I see this try except as a very paranoid check, Updater worthy. The only time I see this might happen is the when updating from 3.9.0, otherwise it should work fine. The functions we're calling here have logging so we'll be able to debug this if we need to.